perses/distributed/feptasks.py

```diff
@@ -206,7 +206,11 @@ def run_protocol(thermodynamic_state, sampler_state, ne_mc_move, topology, n_ite
         Trajectory containing n_iterations frames
 
     """
-    n_atoms = topology.n_atoms
+    #get the atom indices we need to subset the topology and positions
+    atom_indices = topology.select('not water')
```

Comment:
Make the atom indices a parameter to make this tasklet more general. You can do the topology selection just once in the main driver.